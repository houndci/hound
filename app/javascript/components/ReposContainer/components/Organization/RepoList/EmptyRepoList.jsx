import React from 'react';

export default class EmptyRepoList extends React.Component {
  render() {
    return (<ul className="repos"></ul>);
  }
}
