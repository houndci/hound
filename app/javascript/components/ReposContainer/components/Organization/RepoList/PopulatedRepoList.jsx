import React from 'react';

import Repo from './../../Repo';

export default class PopulatedRepoList extends React.Component {
  canShow(repo) {
    const { filterTerm } = this.props;

    if (filterTerm === null) {
      return true;
    } else {
      const repoName = repo.name.toLowerCase();
      return repoName.indexOf(filterTerm.toLowerCase()) !== -1;
    }
  }

  render() {
    const { repos, onRepoClicked, isProcessingId } = this.props;

    return (
      <ul className="repos">
        {repos.filter((repo) => this.canShow(repo)).map( repo => (
          <Repo
            repo={repo}
            key={repo.id}
            onRepoClicked={onRepoClicked}
            isProcessingId={isProcessingId}
          />
        ))}
      </ul>
    );
  }
}
