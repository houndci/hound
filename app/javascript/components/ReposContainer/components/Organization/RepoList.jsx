import React from 'react';

import EmptyRepoList from './RepoList/EmptyRepoList';
import PopulatedRepoList from './RepoList/PopulatedRepoList';

export default class RepoList extends React.Component {
  render() {
    const {
      repos,
      onRepoClicked,
      isProcessingId,
      filterTerm,
    } = this.props;

    if (repos.length > 0) {
      return (
        <PopulatedRepoList
          repos={repos}
          onRepoClicked={onRepoClicked}
          isProcessingId={isProcessingId}
          filterTerm={filterTerm}
        />
      );
    } else {
      return (
        <EmptyRepoList />
      );
    }
  }
}
