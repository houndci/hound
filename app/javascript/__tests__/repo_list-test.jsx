import RepoList from '../components/ReposContainer/components/Organization/RepoList';

it('renders a list of repos appropriately', () => {
  const repos = [
    {
      id: 1,
      name: "Test repo",
      owner: {
        id: 1
      }
    }
  ];

  const onRepoClicked = jest.fn();

  const wrapper = shallow(
    <RepoList
      repos={repos}
      onRepoClicked={onRepoClicked}
      isProcessingId={null}
      filterTerm={""}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
