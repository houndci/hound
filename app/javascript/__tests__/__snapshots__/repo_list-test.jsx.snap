exports[`test renders a list of repos appropriately 1`] = `
<PopulatedRepoList
  filterTerm=""
  isProcessingId={null}
  onRepoClicked={[Function]}
  repos={
    Array [
      Object {
        "id": 1,
        "name": "Test repo",
        "owner": Object {
          "id": 1,
        },
      },
    ]
  } />
`;
